src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -204,6 +204,11 @@ export class SavedSearchEmbeddable
       error: undefined,
     });
 
+    if (this.node) {
+      // to show a loading indicator we need to rerender here
+      this.render(this.node);
+    }
```

Comment:
@kertal Not sure if these indicators are worth an extra render. Otherwise UI does not reflect `searchProps.isLoading` changes.

Reply:
I see! I tested it and think it's an improvement. Especially when the saved search takes a bit longer to reload, the information displayed to the user "Search is still ongoing" is valuable and aligned to e.g. Lens (ideally the embeddable would have an aligned way to showing loading state). But there's no good alternative to a re-render in this case. I try to weight UX (better) and Performance(a bit worse). Personally I think the better UX here is worth the extra render . interested hearing the opinion of the team @elastic/kibana-data-discovery  

And FYI @andreadelrio & @ryankeairns

Reply:
having 👍  @davismcphee , this we're good to proceed.